packages/lit-html/src/lit-html.ts

```diff
@@ -652,6 +654,39 @@ export interface DirectiveParent {
   __directives?: Array<Directive | undefined>;
 }
 
+function trustFromTemplateString(
+  tsa: TemplateStringsArray,
+  stringFromTSA: string
+): TrustedHTML {
+  // A security check to prevent spoofing of Lit template results.
+  // In the future, we may be able to replace this with Array.isTemplateObject,
+  // though we might need to make that check inside of the html and svg
+  // functions, because precompiled templates don't come in as
+  // TemplateStringArray objects.
+  if (!Array.isArray(tsa) || !tsa.hasOwnProperty('raw')) {
```

Comment:
This could be improved to use: https://github.com/google/safevalues/blob/main/src/internals/string_literal.ts

Reply:
True, but we probably don't want to take the dependency or pay for the bytes and runtime overhead.